File: 2022.01.28/javascript/11724.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const [n,m] = input[0].split(' ').map(Number);
let arr = [];
let a,b;
for (let i = 0 ; i<=n ; i++){
  arr.push([]);
}
for (let i = 1 ; i<=m ; i++){
  [a,b] = input[i].split(' ').map(Number);
  arr[a].push(b);
  arr[b].push(a);
}
let visited = new Array(n+1).fill(false);
const dfs = (a) =>{
  if (visited[a]) return 0;
  visited[a] = true;
  for (let i = 0 ; i<arr[a].length ; i++){
    if ( visited[arr[a][i]] ) continue;
    dfs(arr[a][i]);
  }
  return 1;
}
let result = 0;
for (let i = 1 ; i<arr.length ; i++){
  result += dfs(i)
}
console.log(result);

File: 2022.01.28/python/11724.py
import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline
n,m = map(int,input().split())
arr = [[] for _ in range(n+1)]
visited = [False for _ in range(n+1)]
result = 0

for _ in range(m):
    a,b = map(int,input().split())
    arr[a].append(b)
    arr[b].append(a)

def dfs(a):
    if visited[a]: return 0
    visited[a] = True
    for node in arr[a]:
        if visited[node] : continue
        dfs(node)
    return 1

for i in range(1,n+1):
    result += dfs(i)
print(result)